From 464d3cbbd89a5e007fe77ec22f3bb81574be951b Mon Sep 17 00:00:00 2001
From: reichi <reichi@opendreambox.org>
Date: Wed, 5 Jun 2019 10:21:39 +0200
Subject: [PATCH] skin.py: add "global skin variables" define them like this: 
 <globals> 		<value name="infobar_pos" value="0,540" /> 
 </globals>

and then use them like this:
	<screen name="InfoBar" position="{{infobar_pos}}" size="1280,150" backgroundColor="transparent" flags="wfNoBorder">

Change-Id: I543ed2f2a9392e1aec0177e78c03d567cc2383c9
(cherry picked from commit 469a678e29418c14fe1437e217fd661b734da9ef)
---
 usr/lib/enigma2/python/skin.py | 27 +++++++++++++++++++++++----
 1 file changed, 23 insertions(+), 4 deletions(-)

diff --git a/usr/lib/enigma2/python/skin.py b/usr/lib/enigma2/python/skin.py
index f2bcc4c..b9798c8 100644
--- a/usr/lib/enigma2/python/skin.py
+++ b/usr/lib/enigma2/python/skin.py
@@ -13,9 +13,12 @@ from Tools.Import import my_import
 from Tools.LoadPixmap import LoadPixmap
 from Tools.Log import Log
 
+from re import search as re_search
+
 HAS_SKIN_USER_DISPLAY = True
 
 colorNames = dict()
+skinGlobals = dict()
 
 def dump(x, i=0):
 	print " " * i + str(x)
@@ -125,8 +128,18 @@ def getParentSize(desktop, guiObject):
 		parent_size = desktop.size()
 	return parent_size
 
-def parsePosition(str, scale, desktop = None, guiObject = None):
-	p = str.split(',')
+def translateVariable(value):
+	match = re_search("{{(.*)}}", value)
+	if match and len(match.groups()) == 1:
+		key = match.group(1)
+		val = skinGlobals.get(key, value)
+		Log.i("%s => %s" %(key, val))
+		return val
+	return value
+
+def parsePosition(value, scale, desktop = None, guiObject = None):
+	value = translateVariable(value)
+	p = value.split(',')
 	x, y = p[0], p[1]
 	wsize = 1, 1
 	ssize = 1, 1
@@ -150,8 +163,9 @@ def parsePosition(str, scale, desktop = None, guiObject = None):
 
 	return ePoint(x, y)
 
-def parseSize(str, scale, desktop = None, guiObject = None):
-	x, y = str.split(',')
+def parseSize(value, scale, desktop = None, guiObject = None):
+	value = translateVariable(value)
+	x, y = value.split(',')
 	if guiObject is not None:
 		parent_size = getParentSize(desktop, guiObject)
 		#width aliases
@@ -643,6 +657,11 @@ def loadSingleSkinData(desktop, skin, path_prefix):
 		x = eWindowStyleManager.getInstance()
 		x.setStyle(id, style)
 
+	for g in skin.findall("globals"):
+		for value in g.findall("value"):
+			Log.i("Global skin value : %s" %(value.attrib,))
+			skinGlobals[value.attrib["name"]] = value.attrib["value"]
+
 	for components in skin.findall("components"):
 		for component in components.findall("component"):
 			componentSizes.apply(component.attrib)
-- 
2.1.4

